rb/lib/selenium/webdriver/common/interactions/pointer_input.rb

```diff
@@ -0,0 +1,153 @@
+        end
+
+        def encode
+          return nil if no_actions?
+          output = {type: type, id: name, actions: @actions.map(&:encode)}
+          params = {primary: primary, pointerType: kind}
+          output[:parameters] = params
+          output
+        end
+
+        def assert_kind(pointer)
+          raise TypeError, "#{pointer.inspect} is not a valid pointer type" unless KIND.key? pointer
+          KIND[pointer]
+        end
+
+        def create_pointer_move(duration: 0, x: 0, y: 0, element: nil, origin: nil)
+          add_action(PointerMove.new(self, duration, x, y, element, origin))
+        end
+
+        def create_pointer_down(button)
+          add_action(PointerPress.new(self, :down, button))
+        end
+
+        def create_pointer_up(button)
+          add_action(PointerPress.new(self, :up, button))
+        end
+
+        def create_pause(duration = nil)
+          add_action(Pause.new(self, duration))
+        end
+
+        def create_pointer_cancel
+          add_action(PointerCancel.new(self))
+        end
+      end
+
+      class PointerPress < Interaction
+        BUTTONS = {left: 0, middle: 1, right: 2}.freeze
+        DIRECTIONS = {down: :pointerDown, up: :pointerUp}.freeze
+
+        def initialize(source, direction, button)
+          super(source)
+          @direction = assert_direction(direction)
+          @button = assert_button(button)
+        end
+
+        def type
+          @direction
+        end
+
+        def assert_button(button)
+          if button.is_a? Symbol
+            raise ArgumentError, "#{button.inspec} is not a valid button!" unless BUTTONS.key? button
+            button = BUTTONS[button]
+          end
+          raise ArgumentError, 'Button number cannot be negative!' unless button >= 0
+          button
+        end
+
+        def assert_direction(direction)
+          raise TypeError, "#{direction.inspect} is not a valid button direction" unless DIRECTIONS.key? direction
+          DIRECTIONS[direction]
+        end
+
+        def encode
+          {type: type, button: @button}
+        end
+      end # Press
+
+      class PointerMove < Interaction
+        VIEWPORT = :viewport
+        POINTER = :pointer
+        ORIGINS = [VIEWPORT, POINTER].freeze
+
+        def initialize(source, duration, x, y, element, origin)
+          super(source)
+          raise ArgumentError, 'duration value cannot be negative!' unless duration >= 0
+          raise ArgumentError, 'X value cannot be negative!' unless x >= 0
+          raise ArgumentError, 'Y value cannot be negative!' unless y >= 0
+          @duration = duration * DURATION_MULTIPLIER
+          @x_offset = x
+          @y_offset = y
+          @element = element
+          @origin = origin
+        end
+
+        def type
+          :move
+        end
+
+        def assert_direction(direction)
+          raise TypeError, "#{direction.inspect} is not a valid pointer type" unless DIRECTIONS.include? direction
```

Comment:
Indeed, I'll wait a bit for any other reviews and update it accordingly. Thanks!